### PR TITLE
Add `mc-sgx-core-build` crate

### PR DESCRIPTION
**Pull request by nick-mobilecoin**
Add the `mc-sgx-core-build` crate to hold common build utilities used
with other SGX crates.

### CODE REVIEW

File: core/build/README.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
+# Common build logic for SGX libraries
+
+This crate provides common logic to build and link against the Intel SGX SDK
+
+## Table of Contents
+
+- [License](#license)
+- [Build Instructions](#build-instructions)
+- [Intel SGX SDK](#intel-sgx-sdk)
+- [Features](#features)
```
Comment (remoun) [inline core/build/README.md:10]: Stale references
```suggestion
```

File: trusted/trts/sys/Cargo.toml [hunk 2]
```diff
@@ -13,3 +13,4 @@ mc-sgx-core-sys-types = { path = "../../../core/sys/types" }
 [build-dependencies]
 bindgen = "0.60.1"
 cargo-emit = "0.2.1"
+mc-sgx-core-build = { path = "../../../core/build" }
```
Comment (remoun) [inline trusted/trts/sys/Cargo.toml:16]: Nit: Add a newline to maintain separate sorted groups
```suggestion

mc-sgx-core-build = { path = "../../../core/build" }
```

File: untrusted/test_enclave/Cargo.toml [hunk 3]
```diff
@@ -17,3 +17,4 @@ cc = "1.0.73"
 bindgen = "0.60.1"
 rsa = "0.6.1"
 rand = "0.8.5"
+mc-sgx-core-build = { path = "../../core/build" }
```
Comment (remoun) [inline untrusted/test_enclave/Cargo.toml:20]: Nit: Add a newline to maintain separate sorted groups
```suggestion

mc-sgx-core-build = { path = "../../core/build" }
```

File: untrusted/urts/sys/Cargo.toml [hunk 4]
```diff
@@ -14,6 +14,7 @@ mc-sgx-core-sys-types = { path = "../../../core/sys/types" }
 [build-dependencies]
 bindgen = "0.60.1"
 cargo-emit = "0.2.1"
+mc-sgx-core-build = { path = "../../../core/build" }
```
Comment (remoun) [inline untrusted/urts/sys/Cargo.toml:17]: Nit: Add a newline to maintain separate sorted groups
```suggestion

mc-sgx-core-build = { path = "../../../core/build" }
```

File: untrusted/urts/sys/types/Cargo.toml [hunk 5]
```diff
@@ -7,11 +7,10 @@ edition = "2021"
...
+[dev-dependencies]
+test_enclave = { path = "../../../test_enclave" }
 
 [build-dependencies]
 bindgen = "0.60.1"
 cargo-emit = "0.2.1"
-
-[dev-dependencies]
-test_enclave = { path = "../../../test_enclave" }
+mc-sgx-core-build = { path = "../../../../core/build" }
```
Comment (remoun) [inline untrusted/urts/sys/types/Cargo.toml:16]: Nit: Add newlines
```suggestion

mc-sgx-core-build = { path = "../../../../core/build" }

```

**Review by remoun — APPROVED** [review]

File: core/build/README.md (hunk 1 above)
Comment (nick-mobilecoin) [reply core/build/README.md:10]: doh, copy pasta fail, thanks!

**Review by awygle — APPROVED** [review]
LGTM

**Comment by nick-mobilecoin** [comment]
**Graphite Merge Job**

Current status: 🔄 Waiting for checks to pass

This pull request is currently waiting for CI and mergeability checks to pass before being merged.

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/mobilecoinfoundation/sgx/26?utm_source=stack-comment).


Job Reference: QB8STOU63qtQFHJyhfrd

**Pull request merged**